libkbfs/disk_limiter.go

```diff
@@ -1,37 +1,54 @@
 package libkbfs
 
 import (
+	"fmt"
+
 	"github.com/keybase/client/go/protocol/keybase1"
 	"golang.org/x/net/context"
 )
 
-type diskBlockCacheLimiter interface {
-	// onDiskBlockCacheDelete is called by the disk block cache after deleting
-	// blocks from the cache.
-	onDiskBlockCacheDelete(ctx context.Context, blockBytes int64)
+type diskLimitTrackerType int
 
-	// beforeDiskBlockCachePut is called by the disk block cache before putting
-	// a block into the cache. It returns the total number of available bytes.
-	beforeDiskBlockCachePut(ctx context.Context, blockBytes int64) (
-		availableBytes int64, err error)
+type unknownTrackerTypeError struct {
+	typ diskLimitTrackerType
+}
 
-	// afterDiskBlockCachePut is called by the disk block cache after putting
-	// a block into the cache. It returns how many bytes it acquired.
-	afterDiskBlockCachePut(ctx context.Context, blockBytes int64,
-		putData bool)
+func (e unknownTrackerTypeError) Error() string {
+	return fmt.Sprintf("Unknown tracker type: %d", e.typ)
+}
 
-	// onDiskBlockCacheEnable is called when the disk block cache is enabled to
-	// begin accounting for its blocks.
-	onDiskBlockCacheEnable(ctx context.Context, cacheBytes int64)
+const (
+	unknownLimitTracker diskLimitTrackerType = iota
+	journalLimitTracker
+	diskCacheLimitTracker
+	syncCacheLimitTracker
+)
 
-	// onDiskBlockCacheDisable is called when the disk block cache is disabled to
-	// stop accounting for its blocks.
-	onDiskBlockCacheDisable(ctx context.Context, cacheBytes int64)
+type diskLimitByteTracker interface {
```

Comment:
I find that a bit confusing since the journal does actually track bytes -- it just additionally tracks files.  So this name seems like it should apply to the journal tracker as well, until you understand all the details.

Maybe just "simpleByteTracker", with no mention of disks needed, might be better?

Another alternative is changing the interface to be able to track a custom set of values, so it can also be used by the journal tracker.  But that's obviously a lot more work and probably just confuses things more.